### PR TITLE
app-drawer/-layout: Use `click` instead of tap; assignedNodes

### PR DESCRIPTION
**Pull request by keanulee**
`assignedNodes()` doesn't seem to work with Shady DOM. This is causing `app-drawer-layout` tests to fail, even in Canary

### CODE REVIEW

File: app-drawer-layout/app-drawer-layout.html [hunk 1]
```diff
@@ -207,10 +207,12 @@
        * @property drawer
        */
       get drawer() {
-        return Polymer.dom(this).querySelector('[slot=drawer]');
+        return Polymer.Element ?
+            Polymer.dom(this.$.drawerSlot).assignedNodes({ flatten: true })[0] :
+            Polymer.dom(this.$.drawerContent).getDistributedNodes()[0];
```
Comment (frankiefu) [inline app-drawer-layout/app-drawer-layout.html]: Yeah, it doesn't work in Shady DOM... let's do something similar to this for now: https://github.com/PolymerElements/iron-dropdown/blob/2.0-preview/iron-dropdown.html#L350

**Comment by frankiefu** [comment]
I ran the app-drawer-layout/demo and `drawer-toggle` doesn't seem to work in FF.

**Comment by frankiefu** [comment]
Also we should update test/index.html to user `?wc-shadydom=true`

**Comment by keanulee** [comment]
@frankiefu 

Clicking `drawer-toggle` doesn't work due to some bug with Shady DOM (`Polymer.dom(event).localTarget` returns `app-drawer-layout, not the button).

Tests have already been updated.

**Comment by frankiefu** [comment]
LGTM

**Comment by jsilvermist** [comment]
Shouldn't `Polymer.dom(event).localTarget` be changed to `event.target` as stated on https://github.com/polymer/polymer/tree/2.0-preview under scoped events?

**Pull request merged**